crates/scheduler/src/scheduling/batch_scheduler.rs

```diff
@@ -314,12 +280,14 @@ where
                             },
                             timeout: now + Duration::from_secs(10),
                         })
-                    } else if state.sent_updates.contains(&peer_id) {
+                    } else if has_sent_update {
                         ExecutorAction::Train(TrainAction::Idle {
                             timeout: short_idle,
                         })
                     } else if !should_update {
-                        ExecutorAction::Train(TrainAction::ExecuteBatch { batches })
+                        ExecutorAction::Train(TrainAction::ExecuteBatch {
+                            batches: multi_batch_size,
```

Comment:
This doesn't work. We will need to know from the simulation how many batches we can assign to the worker. Think of the example where we have 3 worker with a batch size of 600 and multi_batch_size of 5. If we would like to update after 12000 samples, we assign batches = 5 to each worker in the first round resulting in 9000 processed samples. Now we need to know from the simulation how many batches each worker should process because if we send 5 again we will oversubscribe this round by 6000.

Reply:
LOL, yes I had copied the value from `(false, count, multi_batch_size)` case but should have copied the whole projection, as we need this in multiple places I'm thinking about introducing a `should_execute_batch` helper which considers batch_sizes, progress, stats, and peer_position to determine if any how many batches to execute. WDYT?